main.cpp

```diff
@@ -569,6 +570,11 @@ void do_loop()
 	}
 	os.mqtt.loop();
 
+	if (curr_time > last_subscribe) {
```

Comment:
I initially tried to set up subscriptions in the init function but, if anything happened to the connection, then I would have to resubscribe at some point. That's why I made it a loop, I chose 5 seconds because it would reduce the number of times the controller would have to resubscribe. But the callback makes a lot more sense here. I'll test it out and see if there are any issues.